### ISSUE TITLE
Crash when checking code with bare raise followed by bare except

### ISSUE THREAD

**Issue by pbasista** [PyCQA/pylint#2352]
The current master branch version 436508032373c9c6a876a042c0ab36ede8fc8f63 crashes when checking the following code:

```python
try:
    pass
except RuntimeError:
    raise
except:
    print("a failure")
```

### Current behavior
```
internal error with sending report for module ['test.py']
'NoneType' object has no attribute 'infer'
```

### Expected behavior
No crash and no `try-except-raise` warning reported. This usage of bare raise together with bare except is in my opinion justified and pylint should not warn about using bare raise in this case. It may still warn about using bare except, but that is another topic.

### pylint --version output
```
pylint 2.0.0
astroid 2.0.1
Python 3.6.6 (default, Jun 27 2018, 14:44:17) 
[GCC 8.1.0]
```

The crashing behavior was introduced by PR #2308. It contains a comment identifying the place where the issue occurs.

**Comment by sushobhit27** [comment]
@pbasista I will provide a fix for this.

**Comment by PCManticore** [comment]
#2353 fixed the crash you are getting. Regarding the other comment for `try-except-raise`, I disagree. It's rarely the case that a bare raise together with a bare except are justified, the only case that I could think of is when you want to handle more specialised exception before a general one, but there might be better ways to approach that use case as well.
